src/eq_schema/schema/Questionnaire/index.js

```diff
@@ -0,0 +1,127 @@
+const { last } = require("lodash");
+
+const { SOCIAL } = require("../../../constants/questionnaireTypes");
+const {
+  types: { VOLUNTARY },
+  contentMap
+} = require("../../../constants/legalBases");
+
+const { Confirmation, Introduction, Summary } = require("../../block-types");
+
+const Section = require("../Section");
+const Hub = require("../Hub");
+
+const DEFAULT_METADATA = [
+  {
+    name: "user_id",
+    type: "string"
+  },
+  {
+    name: "period_id",
+    type: "string"
+  },
+  {
+    name: "ru_name",
+    type: "string"
+  }
+];
+
+const SOCIAL_THEME = "social";
+const DEFAULT_THEME = "default";
+const NI_THEME = "northernireland";
+
+const DEFAULT_METADATA_NAMES = DEFAULT_METADATA.map(({ name }) => name);
+
+class Questionnaire {
+  constructor(questionnaireJson) {
+    const questionnaireId = questionnaireJson.id;
+    this.eq_id = questionnaireId;
+    this.form_type = questionnaireId;
+    this.mime_type = "application/json/ons/eq";
+    this.schema_version = "0.0.1";
+    this.data_version = "0.0.3";
+    this.survey_id =
+      questionnaireJson.surveyId ||
+      questionnaireJson.title.toLowerCase().replace(/[^a-z0-9]/g, "");
+    this.title = questionnaireJson.title;
+
+    const ctx = this.createContext(questionnaireJson);
+    // hub will be called here
+    this.hub = this.buildHub(questionnaireJson.hub, ctx);
+    this.sections = this.buildSections(questionnaireJson.sections, ctx);
+    this.buildIntroduction(questionnaireJson.introduction, ctx);
+
+    this.theme =
+      questionnaireJson.type === SOCIAL ? SOCIAL_THEME : DEFAULT_THEME;
+
+    this.theme = questionnaireJson.theme === NI_THEME ? NI_THEME : this.theme;
```

Comment:
We could make a `buildTheme` function below, similar to how `legalBasis` is done. It would make this more tidy.

Reply:
Removed all this as publisher doesn't translate themes as it's not needed. Talked through with @MattElcock